fl4health/utils/load_data.py

```diff
@@ -78,8 +83,29 @@ def load_cifar10_data(
             transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5)),
         ]
     )
+
     training_set: BaseDataset = Cifar10Dataset(data_dir, train=True, transform=transform)
-    validation_set: BaseDataset = Cifar10Dataset(data_dir, train=False, transform=transform)
+    validation_set: BaseDataset
+    if validation_portion == 0:
+        validation_set = Cifar10Dataset(data_dir, train=False, transform=transform)
+    else:
+        validation_set = Cifar10Dataset(data_dir, train=True, transform=transform)
+        dataset_size = len(training_set)
+        validation_size = int(validation_portion * dataset_size)
+        dataset_indexes = list(range(dataset_size))
+        # If hash_key is provided, use it to shuffle the dataset indexes and create a reproducible split
+        if hash_key is not None:
```

Comment:
oh one issue with that is when it is None it doesn't actually uses the existing seed we set and each time it just randomly use a value based in this documentation:

A seed to initialize the [BitGenerator](https://numpy.org/doc/stable/reference/random/bit_generators/generated/numpy.random.BitGenerator.html#numpy.random.BitGenerator). If None, then fresh, unpredictable entropy will be pulled from the OS. If an int or array_like[ints] is passed, then all values must be non-negative and will be passed to [SeedSequence](https://numpy.org/doc/stable/reference/random/bit_generators/generated/numpy.random.SeedSequence.html#numpy.random.SeedSequence) to derive the initial [BitGenerator](https://numpy.org/doc/stable/reference/random/bit_generators/generated/numpy.random.BitGenerator.html#numpy.random.BitGenerator) state. One may also pass in a [SeedSequence](https://numpy.org/doc/stable/reference/random/bit_generators/generated/numpy.random.SeedSequence.html#numpy.random.SeedSequence) instance. Additionally, when passed a [BitGenerator](https://numpy.org/doc/stable/reference/random/bit_generators/generated/numpy.random.BitGenerator.html#numpy.random.BitGenerator), it will be wrapped by [Generator](https://numpy.org/doc/stable/reference/random/generator.html#numpy.random.Generator). If passed a [Generator](https://numpy.org/doc/stable/reference/random/generator.html#numpy.random.Generator), it will be returned unaltered.

Reply:
I will test it again though to see if we can actually collapse it or not.